File: languages/php/Bloock/Entity/Key/ManagedKeyParams.php
<?php

namespace Bloock\Entity\Key;

/**
 * Represents the parameters for creating a managed key.
 */
class ManagedKeyParams
{
    /**
     * Is the name of the managed key.
     * @var string|null
     */
    public ?string $name;
    /**
     * Is the protection level for the key.
     * Is the protection level for the key.
     * @var string
     */
    public string $protection;
    /**
     * Is the timestamp indicating when the key expires.
     * @var string
     */
    public string $keyType;
    /**
     * Is the type of the key.
     * @var int|null
     */
    public ?int $expiration;

    /**
     * Constructs a ManagedKeyParams object with the specified parameters.
     * @param string $protection
     * @param string $keyType
     * @param string|null $name
     * @param int|null $expiration
     */
    public function __construct(string $protection, string $keyType, ?string $name = null, ?int $expiration = null)
    {
        $this->name = $name;
        $this->protection = $protection;
        $this->keyType = $keyType;
        $this->expiration = $expiration;
    }

    public static function fromProto(\Bloock\ManagedKeyParams $res): ManagedKeyParams
    {
        return new ManagedKeyParams(
            $res->getName(),
            $res->getProtection(),
            $res->getKeyType(),
            $res->getExpiration()
        );
    }

    public function toProto(): \Bloock\ManagedKeyParams
    {
        $p = new \Bloock\ManagedKeyParams();
        $p->setProtection(KeyProtectionLevel::toProto($this->protection));
        $p->setKeyType(KeyType::toProto($this->keyType));

        if ($this->name != null) {
            $p->setName($this->name);
        }

        if ($this->expiration != null) {
            $p->setExpiration($this->expiration);
        }
        return $p;
    }
}

File: languages/python/bloock/entity/key/managed_key_params.py
from __future__ import annotations

import bloock._bridge.proto.keys_entities_pb2 as proto
from bloock.entity.key.key_protection_level import KeyProtectionLevel
from bloock.entity.key.key_type import KeyType


class ManagedKeyParams:
    """
    Represents the parameters for creating a managed key.
    """
    def __init__(
            self,
            protection: KeyProtectionLevel,
            key_type: KeyType,
            name: str = "",
            expiration: int = 0,
    ) -> None:
        """
        Constructs a ManagedKeyParams object with the specified parameters.
        :type expiration: object
        :type name: object
        :type key_type: object
        :type protection: object
        :rtype: object
        """
        self.protection = protection
        """
        Is the protection level for the key.
        """
        self.key_type = key_type
        """
        Is the timestamp indicating when the key expires.
        """
        self.name = name
        """
        Is the name of the managed key.
        """
        self.expiration = expiration
        """
        Is the timestamp indicating when the key expires.
        """

    @staticmethod
    def from_proto(key: proto.ManagedKeyParams) -> ManagedKeyParams:
        return ManagedKeyParams(
            protection=KeyProtectionLevel.from_proto(key.protection),
            key_type=KeyType.from_proto(key.key_type),
            name=key.name,
            expiration=key.expiration
        )

    def to_proto(self) -> proto.ManagedKeyParams:
        return proto.ManagedKeyParams(
            protection=self.protection.to_proto(),
            key_type=self.key_type.to_proto(),
            name=self.name,
            expiration=self.expiration
        )
